src/lib/mina_lib/mina_lib.ml

```diff
@@ -778,10 +780,89 @@ let last_epoch_delegators t ~pk =
   in
   find_delegators last_epoch_delegatee_table pk
 
+let perform_compaction t =
+  if
+    not
+      (Genesis_constants.Proof_level.equal
+         t.config.precomputed_values.proof_level Full)
```

Comment:
Is there a reason not to perform compaction for other proof levels? If so, could you add a comment mentioning why?

Reply:
I did not want to add this to the test profiles where we do timing based checks and calling compact multiple times might impact that?  Time taken by compact () is almost the same as slot duration on test profiles with proof level `check` and `none`.  Maybe it would be better to have a new value in the mlh files to determine this?